src/osd/PrimaryLogPG.cc

```diff
@@ -769,6 +796,17 @@ class PGLSParentFilter : public PGLSFilter {
                       bufferlist& outdata) override;
 };
 
+bool PGLSNameFilter::filter(const hobject_t &obj,
```

Comment:
would be better to move the implementation of this method up to where it is declared instead of interleaving it with another class.